File: hej/apps/hej-web/lib/api/projects.ts
/**
 * Projects API module.
 *
 * Wraps backend endpoints:
 *   GET /organizations/{org_id}/projects   (list by org)
 *   GET /projects/{project_id}             (single project)
 *
 * Note: backend has no top-level `GET /projects`, so `listProjects` takes
 * an `orgId`. Listing across orgs is the caller's responsibility (do
 * `listOrganizations()` first and iterate).
 */
import { apiClient, type ApiResult } from "./client"

export type ApiProject = {
  id: string
  organization_id: string
  name: string
  description: string | null
  governance_model: string
  status: string
  created_at: string
  created_by: string | null
  updated_at: string | null
  updated_by: string | null
}

export type CreateProjectInput = {
  name: string
  description: string | null
  governance_model: string
}

export function listProjects(orgId: string): Promise<ApiResult<ApiProject[]>> {
  return apiClient.get<ApiProject[]>(`/organizations/${orgId}/projects`)
}

export function getProject(projectId: string): Promise<ApiResult<ApiProject>> {
  return apiClient.get<ApiProject>(`/projects/${projectId}`)
}

export function createProject(
  orgId: string,
  payload: CreateProjectInput
): Promise<ApiResult<ApiProject>> {
  return apiClient.post<ApiProject>(`/organizations/${orgId}/projects`, payload)
}

File: hej/apps/hej-api/app/schemas/projects.py
from datetime import datetime

from pydantic import BaseModel

from app.models import GovernanceModel
from app.models import ProjectStatus


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None
    governance_model: GovernanceModel = GovernanceModel.STANDARD


class ProjectRead(BaseModel):
    id: str
    organization_id: str
    name: str
    description: str | None
    governance_model: GovernanceModel
    status: ProjectStatus
    created_at: datetime
    created_by: int | None = None
    updated_at: datetime | None = None
    updated_by: int | None = None
